src/main.rs

```diff
@@ -1,4 +1,146 @@
+    task::block_on(mint_call.send()).expect("Failed to send mint transaction");
+
+    // Query Total Supply
+    let total_supply = nft_contract.method::<_, Uint>("totalSupply", ())
+        .expect("Error finding method").call().await.expect("Error sending total supply call");
+    println!("Total supply: {}", total_supply);
+
+    // Query royalty info
+    let token_id = ethers::core::types::U256::from_dec_str("1").expect("Couldn't convert 1 to U256");
+    let sale_price = ethers::core::types::U256::from_dec_str("5000000000000000000").expect("Couldn't convert 5*10^18 to U256");
+    let royalty_info = nft_contract.method::<_, (ethers::core::types::Address, ethers::core::types::U256)>("royaltyInfo", (token_id, sale_price))
+        .expect("Error finding royaltyInfo method").call().await.expect("Error sending royaltyInfo call");
+
+
+    println!("Royalty info: address: {}, amount: {}", royalty_info.0, royalty_info.1);
+
+    // List
+    let provider = Provider::<Http>::try_from(anvil.endpoint()).expect("Failed to connect to Anvil").interval(Duration::from_millis(10u64));
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet.with_chain_id(anvil.chain_id())));
+    let zora_address: ethers::core::types::Address = "0x5f7072E1fA7c01dfAc7Cf54289621AFAaD2184d0".parse::<Address>().expect("Couldn't parse zora address");
+    let transfer_helper_address: H160 = "0x909e9efE4D87d1a6018C2065aE642b6D0447bc91".parse::<Address>().expect("Couldn't parse transfer helper address");
+    let module_manager_address: H160 = "0x850A7c6fE2CF48eea1393554C8A3bA23f20CC401".parse::<Address>().expect("Couldn't parse transfer helper address");
+
+
+    // Approve
+    let reserve_auction_contract :Contract<SignerMiddleware<ethers_providers::Provider<Http>, Wallet<ethers::core::k256::ecdsa::SigningKey>>> = Contract::new(zora_address, RESERVEAUCTIONCOREETH_ABI.clone(), client.clone());
+    let approve_call = nft_contract.set_approval_for_all(transfer_helper_address, true);
+    task::block_on(approve_call.send()).expect("Failed to send approve transaction");
+
+    let module_manager_contract :Contract<SignerMiddleware<ethers_providers::Provider<Http>, Wallet<ethers::core::k256::ecdsa::SigningKey>>> = Contract::new(module_manager_address, MODULEMANAGER_ABI.clone(), client.clone());
+    module_manager_contract.method::<_, ()>("setApprovalForModule", (zora_address, true))
+        .expect("Error finding setApprovalForModule method").send().await.expect("Error calling approval for module");
+
+    // Create auction
+    println!("Creating listing");
+    let auction_duration: ethers::prelude::U256 = ethers::core::types::U256::from(3*24*3600); // 3 days
+    let auction_reserve_price: ethers::prelude::U256 = ethers::core::types::U256::from_dec_str("10000000000000000").expect("Reserve price should be a valid number");
```

Comment:
Not blocking, why not make these numbers more readable in all occurrences (like you do with 3 days), so for example this could be `from_dec_str("0.01*10^18") // 0.01 ether`